service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ObservationStatementMapper.java

```diff
@@ -114,4 +164,12 @@ private String prepareCode(Observation observation) {
     private boolean isRangeUnitValid(String unit, Quantity quantity) {
         return quantity.hasUnit() && !unit.equals(quantity.getUnit());
     }
+
+    private boolean isInterpretationCode(Coding coding) {
+        String codingSystem = coding.getSystem();
+        String code = coding.getCode();
+
+        return (coding.hasSystem() && codingSystem.equals(INTERPRETATION_CODE_SYSTEM))
+            && Set.of("H", "HH", "HU", "L", "LL", "LU", "A", "AA").contains(code);
```

Comment:
Does this set need to be created every time this method is called?

Reply:
Interesting; I assume(d) the jvm recognises it as constant and optimises it, but I can't find anything concrete about it - and of course it could be vendor specific.

I agree it may be best to extract this as a class constant.

Reply:
Extracted into a class constant